packages/components/src/TableToolbar/TableToolbar.tsx

```diff
@@ -1,23 +1,40 @@
 import React, { Fragment } from 'react';
-import { Toolbar } from '@patternfly/react-core';
+import { Toolbar, ToolbarProps } from '@patternfly/react-core';
 import { useOUIAId } from '@patternfly/react-core/';
 import classNames from 'classnames';
-import propTypes from 'prop-types';
 
 import './TableToolbar.scss';
 
-function generateCount(results) {
+export interface TableToolbarProps extends Omit<Omit<ToolbarProps, 'selected'>, 'ref'> {
+  isFooter?: boolean;
+  results?: number;
+  className?: string;
+  selected?: number;
+  ouiaId?: string;
+  ouiaSafe?: boolean;
+}
+
+function generateCount(results: number) {
   if (results > 1 || results < 1) {
     return `${results} Results`;
   } else {
     return `${results} Result`;
   }
 }
 
-const TableToolbar = ({ isFooter, results, className, selected, children, ouiaId, ouiaSafe = true, ...props }) => {
+const TableToolbar: React.FunctionComponent<TableToolbarProps> = ({
+  isFooter = false,
+  results = 0,
+  className,
+  selected = 0,
+  children,
+  ouiaId,
+  ouiaSafe = true,
+  ...props
+}) => {
   const tableToolbarClasses = classNames('ins-c-table__toolbar', { [`ins-m-footer`]: isFooter }, className);
   const ouiaComponentType = 'RHI/TableToolbar';
-  const ouiaFinalId = useOUIAId(ouiaComponentType, ouiaId, ouiaSafe);
+  const ouiaFinalId = useOUIAId(ouiaComponentType, ouiaId, ouiaSafe as unknown as string);
```

Comment:
As uknown as string, is that correct?

Reply:
Yeah. I think this is a typings issue in PF. the `ouiaSafe` is required to be `boolean` by the component. However, the `useOUIAId` requires the last argument to be a string.

Reply:
> I think we should just start removing deprecated components instead of convertim them to ts.
> 
> Breadcrumbs in PF exist for years now.

And yeah I would like to do a major audit of FEC during the tech debt week.